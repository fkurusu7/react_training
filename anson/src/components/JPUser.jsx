import { useContext } from "react";
import UserContext from "../utils/context/userContext";

function JPUser() {
  const userCtxData = useContext(UserContext);

  return (
    <div>
      <>
        <span>ID: {userCtxData.id}</span>
        <h2>Name: {userCtxData.name}</h2>
        <p>Username: {userCtxData.username}</p>
        <p>Email: {userCtxData.email?.toLowerCase()}</p>
      </>

      {/* {error && <p>{error}</p>} */}
    </div>
  );
}

export default JPUser;
