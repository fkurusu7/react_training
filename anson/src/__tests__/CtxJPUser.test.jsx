import { render, screen } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import JPUser from "../components/JPUser";
import UserContext from "../utils/context/userContext";

describe("render context values", () => {
  const mockUserCtxData = {
    id: 1000,
    name: "Johny",
    username: "jj",
    email: "j@gmail.com",
    setUser: () => {},
  };

  it("should match snapshot", () => {
    const { container } = render(
      <UserContext.Provider value={mockUserCtxData}>
        <JPUser />
      </UserContext.Provider>
    );

    expect(container).toMatchSnapshot();
  });

  it("should display correct user name", () => {
    render(
      <UserContext.Provider value={mockUserCtxData}>
        <JPUser />
      </UserContext.Provider>
    );

    expect(screen.queryByText("Name: Johny")).exist;
  });
});
