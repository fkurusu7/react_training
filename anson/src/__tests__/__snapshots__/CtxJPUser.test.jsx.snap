// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`render context values > should match snapshot 1`] = `
<div>
  <div>
    <span>
      ID: 
      1000
    </span>
    <h2>
      Name: 
      Johny
    </h2>
    <p>
      Username: 
      jj
    </p>
    <p>
      Email: 
      j@gmail.com
    </p>
  </div>
</div>
`;
